### DISCUSSION TITLE
Form drawers

### DISCUSSION THREAD

**Discussion by alexander-bobin** [reapit/elements#61] [General]
@willmcvay one thing we see commonly is what we call "form drawers". I suspect those would come up quite commonly in your requirements too.

I understand the `Drawer` component was recently added to `Elements`. Have you or any consumers used this component to deliver a form drawer. If so, can we see an example of how it was done in code?

**Comment by adrian-reapit** [comment]
Hi @alexander-bobin 

We are just starting to add forms to drawers. I could talk you through what we have done. The main modification to normal layout is to force PersistentNotification, which we use as an error summary, expand to fill the width of the drawer

<img width="1384" alt="image" src="https://github.com/reapit/elements/assets/152165863/41d50a42-ed83-4d0a-9fdb-49b3b04a4383">

**Reply by alexander-bobin** [reply]
@adrian-reapit a talk through would be really useful. One thing we see in the design system is form buttons being placed in the footer. In the past this has lead us towards using a wrapping form (although not 100% necessary).

This image is just an example, but placing buttons in the footer seems to be locked in.
<img width="475" alt="Screenshot 2024-03-20 at 8 59 47 am" src="https://github.com/reapit/elements/assets/1836301/acae04de-3216-4bf0-a92c-101425d35ee7">

Anyways, let's organise a time for a chat and compare notes.

**Comment by willmcvay** [comment]
@alexander-bobin The Drawer component accepts children so you can put a form or any other component within it as per above see: https://github.com/reapit/foundations-react-vite-template/blob/master/src/components/contacts/list/index.tsx#L330 for an example - in this case, the <ContactQuickView /> would be your form. Hope this makes sense

**Comment by alexander-bobin** [comment]
Closing this discussion. I had a chat with Dimas and get a better sense of how the Drawer component would be used to deliver form drawers.
